rdflib/namespace.py

```diff
@@ -330,6 +330,21 @@ def compute_qname(self, uri, generate=True):
             namespace, name = split_uri(uri)
             namespace = URIRef(namespace)
             prefix = self.store.prefix(namespace)
+            # check to see if there are longer prefixes defined
+            temp_namespace = namespace.toPython()
+            output_prefix = prefix
+            output_namespace = namespace
+            output_name = name
+            for i in xrange(1, len(name) + 1):
```

Comment:
we really cannot have this loop here. This function gets called for every single URI in the store on serialisation. 

You can have a loong URI and this will call the store for each substring in that URL. For some stores, this means network activity (sqlalchemy, etc). 

If we really want to fix this, the namespace manager needs to index the available prefixes by length in __init__ and then try the longest first. But even this means many "if uri starts with x" calls. Some sort of trie-data-structure may be useful here.